client/allocrunner/groupservice_hook.go

```diff
@@ -3,30 +3,65 @@ package allocrunner
 import (
 	"sync"
 
-	hclog "github.com/hashicorp/go-hclog"
 	log "github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/nomad/client/allocrunner/interfaces"
 	"github.com/hashicorp/nomad/client/consul"
+	"github.com/hashicorp/nomad/client/taskenv"
+	agentconsul "github.com/hashicorp/nomad/command/agent/consul"
 	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/plugins/drivers"
 )
 
 // groupServiceHook manages task group Consul service registration and
 // deregistration.
 type groupServiceHook struct {
-	alloc        *structs.Allocation
+	allocID      string
+	group        string
+	restarter    agentconsul.WorkloadRestarter
 	consulClient consul.ConsulServiceAPI
 	prerun       bool
-	mu           sync.Mutex
 
 	logger log.Logger
+
+	// The following fields may be updated
+	canary   bool
+	services []*structs.Service
+	networks structs.Networks
```

Comment:
Can networks be updated inplace? We can dynamically add/remove ports and port mappings from a running task?

Reply:
Huh, I guess the old task services code handled network updates... I'm still not sure that's necessary.